experiments/multi_prompting/chat.py

```diff
@@ -0,0 +1,85 @@
+import os
+import streamlit as st
+from dotenv import load_dotenv
+from langchain.chains import ConversationChain
+from langchain.chat_models import ChatOpenAI
+from langchain.prompts import (
+    ChatPromptTemplate, 
+    MessagesPlaceholder, 
+    SystemMessagePromptTemplate, 
+    HumanMessagePromptTemplate
+)
+
+load_dotenv()
+os.environ["OPENAI_API_KEY"] = os.getenv('API_KEY')
+
+# chooses which color to display percentage as based on value and displays it
+def display_percentage(percentage):
+    if percentage <= 20:
+        percentage = str(percentage)
+        st.write( f":red[{percentage}%]")
+
+    elif percentage > 20 and percentage <= 50:
+        percentage = str(percentage)
+        st.write( f":blue[{percentage}%]")
+
+    elif percentage > 50:
+        percentage = str(percentage)
+        st.write( f":green[{percentage}%]")
+
+
+# creates a prompt for the ai to use and responds to the user's inquiry
+def get_response_with_memory(model, user_inquiry, memory):
+    prompt = ChatPromptTemplate.from_messages([
+        SystemMessagePromptTemplate.from_template("You are a helpful chatbot"),
+        MessagesPlaceholder(variable_name="history"),
+        HumanMessagePromptTemplate.from_template("{input}")
+    ])
+
+    conversation = ConversationChain(memory=memory, 
+                                    prompt=prompt,
+                                    verbose= True, 
+                                    llm=ChatOpenAI(temperature=0.6, model= model),)
+    
+    response = conversation.predict(input=user_inquiry)
+
+    return response
+
+
+# makes the prompt for the AI to follow
+def make_prompt(first_prompt, second_prompt, third_prompt):
```

Comment:
you could name the first_prompt, second_prompt and third_prompt arguments with a more meaningful description of the use of these... if the first_prompt is like "initial_condition". IDK exactly what are the steps of the inner monologue but the idea is to describe each three. The goal is that other developers understand the context of why we use 3 arguments and not 4 etc...

Reply:
after reading all the code I believe the specification of what the prompt1, 2, and 3 are, is not defined. In this case, we are combining the three prompts. Let's discuss what is the main purpose of the 3 prompts. :)